livekit-agents/livekit/agents/llm/chat_context.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2023 LiveKit, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+import enum
+from dataclasses import dataclass, field
+from typing import Tuple
+
+from livekit import rtc
+
+
+class ChatRole(enum.Enum):
+    SYSTEM = "system"
+    USER = "user"
+    ASSISTANT = "assistant"
+    TOOL = "tool"
+
+
+@dataclass
+class ChatMessage:
+    role: ChatRole
+    text: str
+    images: list[ChatImage] = field(default_factory=list)
+
+
+@dataclass
+class ChatContext:
+    messages: list[ChatMessage] = field(default_factory=list)
+
+
+@dataclass
+class ChatImage:
+    image: str | rtc.VideoFrame
+    dimensions: Tuple[int, int]
```

Comment:
do we want a default here?

Reply:
dimensions is confusing with the width and height already encoded inside rtc.VideoFrame.
We should rename dimensions to inference_dimenstions? 
@keepingitneil @davidzhao 

Also using a tuple is less explicit in situations you're not using typing. not a strong opinion but I tend to prefer having two separate fields. Not a big deal since ChatImage is a struct anyway

Reply:
re, it makes stuff hard to read. e.g: https://github.com/livekit/agents/blob/cf3c5344958b3658d577c4c57fb909dfe41b39a7/livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/llm.py#L296

Reply:
inference_width
inference_height 

I think these may be better, also let's default to None the values